test/contracts/kill_switch/kill_switch.test.js

```diff
@@ -0,0 +1,963 @@
+        context('when there is no bug registered', () => {
+          context('when there is no highest allowed severity set for the contract being called', () => {
+            it('returns true', async () => {
+              assert.isTrue(await killSwitch.checkSeverityIssuesFor(SAMPLE_APP_ID, appBase.address))
+            })
+          })
+
+          context('when there is a highest allowed severity set for the contract being called', () => {
+            beforeEach('set highest allowed severity', async () => {
+              await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.LOW, { from: owner })
+            })
+
+            it('returns true', async () => {
+              assert.isTrue(await killSwitch.checkSeverityIssuesFor(SAMPLE_APP_ID, appBase.address))
+            })
+          })
+        })
+
+        context('when there is a bug registered', () => {
+          beforeEach('register a bug', async () => {
+            await defaultIssuesRegistry.setSeverityFor(appBase.address, SEVERITY.MID, { from: securityPartner })
+          })
+
+          context('when there is no highest allowed severity set for the contract being called', () => {
+            it('returns false', async () => {
+              assert.isFalse(await killSwitch.checkSeverityIssuesFor(SAMPLE_APP_ID, appBase.address))
+            })
+          })
+
+          context('when there is a highest allowed severity set for the contract being called', () => {
+            context('when the highest allowed severity is under the reported bug severity', () => {
+              beforeEach('set highest allowed severity', async () => {
+                await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.LOW, { from: owner })
+              })
+
+              it('returns false', async () => {
+                assert.isFalse(await killSwitch.checkSeverityIssuesFor(SAMPLE_APP_ID, appBase.address))
+              })
+            })
+
+            context('when the highest allowed severity is equal to the reported bug severity', () => {
+              beforeEach('set highest allowed severity', async () => {
+                await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.MID, { from: owner })
+              })
+
+              it('returns true', async () => {
+                assert.isTrue(await killSwitch.checkSeverityIssuesFor(SAMPLE_APP_ID, appBase.address))
+              })
+            })
+
+            context('when the highest allowed severity is greater than the reported bug severity', () => {
+              beforeEach('set highest allowed severity', async () => {
+                await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.CRITICAL, { from: owner })
+              })
+
+              it('returns true', async () => {
+                assert.isTrue(await killSwitch.checkSeverityIssuesFor(SAMPLE_APP_ID, appBase.address))
+              })
+            })
```

Comment:
Perhaps we can also write a test or two here that checks that `checkSeverityIssuesFor()` reflects the latest issue registry set if the specific app gets set to use a different issues registry (or the `KillSwitch` sets a different default registry).